app/controllers/index.rb

```diff
@@ -0,0 +1,164 @@
+#TODO: -stretch goal- redirect back to previous page if you go to login or register. 
+
+#User.cars or User.prices gets cars or prices for that instance of user. 
+
+
+get '/' do
+	redirect '/cars'
+end
+
+get '/login' do
+	redirect '/cars' if logged_in?
+	
+	erb :login
+end
+
+#authenticates login
+post '/login' do
+	@email = params[:email]
+	@password = params[:password]
+	@user = user_at_email(params[:email])
+	
+	if fields_not_blank?( @email, @password )
+		if user_is_authenticated?( @user, @password )
+			session_set_current_user(@user)
+			redirect '/cars'
+		else
+			@login_error = "The email address or password you’ve entered is not valid."
+			erb :login
+		end
+	else
+		@login_error = "Please enter all required fields"
+		erb :login
+	end
+end
+
+# display register page
+get '/register' do
+	@user = nil
+	redirect '/cars' if logged_in?
+	erb :register
+end
+
+# goes to homepage or back to register page
+post '/register' do
+	@name = params[:name]
+	@email = params[:email]
+	@password = params[:password]
+	if any_blank_fields?( @email, @password, @name)
+		@errors = "Please enter all required fields"
+		erb :register
+	else
+		@user = User.create(name: @name, email: @email, password: @password)
+		if @user.errors[:email].any?
```

Comment:
You're also only checking for errors on the email attribute and potentially ignoring other errors.